timemachine/fe/mle.py

```diff
@@ -212,3 +215,60 @@ def infer_node_vals_and_errs(
     dg_err = bootstrap_estimates.std(0)
 
     return dg, dg_err
+
+
+def infer_node_vals_and_errs_networkx(
+    nx_graph: nx.Graph,
+    edge_diff_prop: str,
+    edge_stddev_prop: str,
+    ref_nodes: Sequence[Any],
+    ref_node_val_prop: str,
+    ref_node_stddev_prop: Optional[str] = None,
+    n_bootstrap: int = 100,
+    seed: int = 0,
+) -> Tuple[NDArray, NDArray]:
+    """Version of :py:func:`fe.mle.infer_node_vals_and_errs` that accepts a networkx graph.
+
+    Parameters
+    ----------
+    nx_graph: nx.Graph
+        Networkx graph
+    edge_diff_prop: str
+        Edge property to use for differences
+    edge_stddev_prop: str
+        Edge property to use for standard deviations
+    ref_nodes: sequence
+        Reference nodes (must have properties ref_node_val_prop and ref_node_stddev_prop)
+    ref_node_val_prop: str
+        Node property to use for reference values
+    ref_node_stddev_prop: str or None, optional
+        Node property to use for reference standard deviations. If None, assume zero uncertainty in reference values.
+    n_bootstrap, seed:
+        See documentation for :py:func:`fe.mle.infer_node_vals_and_errs`
+
+    Returns
+    -------
+    dg: [K] array
+        inferred absolute values
+    dg_err : [K] array
+    """
+
+    node_to_idx = {n: idx for idx, n in enumerate(nx_graph.nodes)}
```

Comment:
Looking at the [nx.relabel_nodes](https://networkx.org/documentation/stable/reference/generated/networkx.relabel.relabel_nodes.html#networkx.relabel.relabel_nodes) documentation, looks like it can change the ordering of the nodes. Which probably makes @maxentile suggestion important to ensure users know exactly which node the dG is assigned to.

Reply:
Nice, I like these suggestions. Agreed that (2) seems particularly nice. Implemented in https://github.com/proteneer/timemachine/pull/987/commits/9a5f0defe9b59cb04ce34feee4184bdd1665b351